### PR TITLE
update cluster status conditions when cluster status controller stopped posting cluster status

### PR DESCRIPTION
**Pull request by xigang**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
update cluster status conditions when cluster status controller stopped posting cluster status

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by XiShanYongYe-Chang** [comment]
Hi @xigang, can you explain the problem briefly?

File: pkg/controllers/cluster/cluster_controller.go [hunk 1]
```diff
@@ -543,6 +543,7 @@ func (c *Controller) tryUpdateClusterHealth(ctx context.Context, cluster *cluste
 					currentCondition.Reason = "ClusterStatusUnknown"
 					currentCondition.Message = "Cluster status controller stopped posting cluster status."
 					currentCondition.LastTransitionTime = nowTimestamp
+					meta.SetStatusCondition(&cluster.Status.Conditions, *currentCondition)
```
Comment (RainbowMango) [inline pkg/controllers/cluster/cluster_controller.go:546]: I think the condition of `cluster` could be updated by the pointer(`currentCondition`), so we might don't need to set the conditions explicitly.

cc author @Garrybest to confirm.

Comment (RainbowMango) [reply pkg/controllers/cluster/cluster_controller.go:546]: > currentCondition := meta.FindStatusCondition([]metav1.Condition{initCondition}, "Ready")

I guess the `[]metav1.Condition{initCondition}` just builds a new slice, which means `currentCondition` essentially points to a copy of `initCondition`. So changes made on `currentCondition` won't reflect to `initCondition`.

Comment (RainbowMango) [reply pkg/controllers/cluster/cluster_controller.go:546]: Could you please try the following sample? 
```golang
package main

import (
	"encoding/json"
	"fmt"

	"k8s.io/apimachinery/pkg/api/meta"
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
)

func main() {
	initCondition := metav1.Condition{
		Type:               "Ready",
		Status:             metav1.ConditionTrue,
		ObservedGeneration: 123,
		Reason:             "Reason",
		Message:            "Message",
	}
        conditions := []metav1.Condition{initCondition} // build a new slice to simulate 'cluster.Status.Conditions'
	currentCondition := meta.FindStatusCondition(conditions, "Ready") // simulates meta.FindStatusCondition(cluster.Status.Conditions, clusterConditionType)
	currentCondition.Status = metav1.ConditionUnknown
	currentCondition.Reason = "ClusterStatusUnknown"
	currentCondition.Message = "Cluster status controller stopped posting cluster status."

	newData, _ := json.MarshalIndent(currentCondition, " ", " ")
	fmt.Printf("newData: %v\n\n", string(newData))

	oldData, _ := json.MarshalIndent(conditions, " ", " ") // dump conditions, the old conditions should be updated as well
	fmt.Printf("oldData: %v\n", string(oldData))
}
```

Comment (xigang) [reply pkg/controllers/cluster/cluster_controller.go:546]: @RainbowMango from the program output to see the code is no problem:)

program output:
```
	newData: {
	  "type": "Ready",
	  "status": "Unknown",
	  "observedGeneration": 123,
	  "lastTransitionTime": null,
	  "reason": "ClusterStatusUnknown",
	  "message": "Cluster status controller stopped posting cluster status."
	 }
	
	oldData: [
	  {
	   "type": "Ready",
	   "status": "Unknown",
	   "observedGeneration": 123,
	   "lastTransitionTime": null,
	   "reason": "ClusterStatusUnknown",
	   "message": "Cluster status controller stopped posting cluster status."
	  }
	 ]
```

**Pull request closed without merging**